lib/collection.dart

```diff
@@ -77,3 +77,14 @@ bool setsEqual(Set a, Set b) {
 
   return a.containsAll(b);
 }
+
+/// Returns the index of the first item in [list] where [matcher] evaluates to
+/// true.
+///
+/// Returns -1 if there are no items where [matcher] evaluates to true.
+int indexOf<T>(List<T> list, bool matcher(T item)) {
```

Comment:
I think a more common name for the callback is `predicate`. `matcher` sounds like something from `package:matcher`. Similarly, a more common name for items in a collection is `element`.